Remove bad param when setting OPTIMADE response fields

ml-evs opened the pull request:
## Summary

OPTIMADE uses the `response_fields` URL parameter to select the response fields, this PR removes an extraneous URL parameter that is added when using a raw OPTIMADE filter through `OptimadeRester`.

Addresses and closes #2243 (cc @mkhorton)

~Haven't added any tests as I don't think this extension is being tested at the moment~

## Checklist

Work-in-progress pull requests are encouraged, but please put [WIP]
in the pull request title.

Before a pull request can be merged, the following items must be checked:

- [x] Code is in the [standard Python style](https://www.python.org/dev/peps/pep-0008/). The easiest way to handle this
      is to run the following in the **correct sequence** on your local machine. Start with running
      [black](https://black.readthedocs.io/en/stable/index.html) on your new code. This will automatically reformat
      your code to PEP8 conventions and removes most issues. Then run 
      [pycodestyle](https://pycodestyle.readthedocs.io/en/latest/), followed by 
      [flake8](http://flake8.pycqa.org/en/latest/).
- [x] Docstrings have been added in the [Google docstring format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).
      Run [pydocstyle](http://www.pydocstyle.org/en/2.1.1/index.html) on your code.
- [x] Type annotations are **highly** encouraged. Run [mypy](http://mypy-lang.org/) to type check your code.
- [x] Tests have been added for any new functionality or bug fixes.
- [x] All linting and tests pass.

mkhorton commented:
Thanks @ml-evs! Since you're a first time contributor, I had to manually approve the CI, but will merge once it's run.

Btw, (also since you're a first time contributor), please make sure to fill out [this form](https://forms.gle/JnisFb38QDR8QTFTA) so we can acknowledge you appropriately on our [development team page](https://pymatgen.org/team.html).

ml-evs commented:
Hi @mkhorton, looks like I need approval again for the workflows.

I've rebased my PR on master so now the `retrying` dep failures should not occur. I also added a test for the raw filter.

mkhorton commented:
Much obliged @ml-evs !
